includes/template-library/manager.php

```diff
@@ -766,10 +777,22 @@ private function is_allowed_to_read_template( array $args ): bool {
 			$this->set_wordpress_adapter( new WordPress_Adapter() );
 		}
 
+		if ( $this->is_global_widget( $args ) ) {
+			return true;
+		}
+
 		$post_id = intval( $args['template_id'] );
 		$post_status = $this->wordpress_adapter->get_post_status( $post_id );
 		$is_private_or_non_published = ( 'private' === $post_status && ! $this->wordpress_adapter->current_user_can( 'read_private_posts', $post_id ) ) || ( 'publish' !== $post_status );
 
 		return $is_private_or_non_published || $this->wordpress_adapter->current_user_can( 'edit_post', $post_id );
 	}
+
+	private function is_global_widget( array $args ): bool {
+		if ( null === $this->elementor_adapter ) {
+			$this->set_elementor_adapter( new Elementor_Adapter() );
+		}
+
+		return isset( $args['global_widget'] ) || 'widget' === $this->elementor_adapter->get_template_type( $args['template_id'] );
```

Comment:
Is checking the template type not enough? Adding the `global_widget` arg is causing a version dependancy.

Reply:
Args is something I added now to pro
I want to avoid db calls on each iteration
`'widget' === $this->elementor_adapter->get_template_type( $args['template_id'] )` should be removed on 3.28
I'll add a comment.